datatree/__init__.py

```diff
@@ -1,10 +1,20 @@
+import warnings
+
 # import public API
 from .datatree import DataTree
 from .extensions import register_datatree_accessor
 from .io import open_datatree
 from .mapping import TreeIsomorphismError, map_over_subtree
 from .treenode import InvalidTreeError, NotFoundInTreeError
 
+warnings.warn(
+    "The repository `xarray-contrib/datatree` has been archived! It is no longer maintained as of October 2024. "
+    "Datatree functionality has been migrated upstream to `pydata/xarray`, so please use `xarray.DataTree` instead. "
+    "To help you migrate there is a guide at https://github.com/pydata/xarray/blob/main/DATATREE_MIGRATION_GUIDE.md",
```

Comment:
this might be potentially confusing, since we do plan on moving it out of the `xarray` repository at some point. Should we instead open and point to a dedicated discussion page here that we can more easily modify?